File: frontend/src/types/application.ts
export type ApplicationStatus =
  | "draft"
  | "applied"
  | "replied"
  | "interview"
  | "offer"
  | "rejected"
  | "ghosted";

/** Canonical pipeline LOV — string values must match api-gateway validation (see docs/IMPLEMENTATION-GUIDE.md Appendix A). */
export const APPLICATION_PIPELINE_STATUS_OPTIONS: {
  value: ApplicationStatus;
  label: string;
}[] = [
  { value: "draft", label: "Draft" },
  { value: "applied", label: "Applied" },
  { value: "replied", label: "Replied" },
  { value: "interview", label: "Interview" },
  { value: "offer", label: "Offer" },
  { value: "rejected", label: "Rejected" },
  { value: "ghosted", label: "Ghosted" },
];

export interface Application {
  id: string;
  user_id: string;
  resume_id: string | null;
  company_name: string;
  role: string;
  recruiter_email: string;
  job_description: string;
  job_link: string;
  status: ApplicationStatus;
  created_at: string;
  updated_at: string;
}

export interface CreateApplicationRequest {
  company_name: string;
  role: string;
  recruiter_email: string;
  job_description: string;
  job_link: string;
  resume_id?: string;
}

export interface ApplicationResponse {
  application: Application;
}

export interface ApplicationListResponse {
  applications: Application[];
}

File: api-gateway/internal/model/application.go
package model

import "time"

type Application struct {
	ID             string    `json:"id"`
	UserID         string    `json:"user_id"`
	ResumeID       *string   `json:"resume_id,omitempty"`
	CompanyName    string    `json:"company_name"`
	Role           string    `json:"role"`
	RecruiterEmail string    `json:"recruiter_email,omitempty"`
	JobDescription string    `json:"job_description,omitempty"`
	JobLink        string    `json:"job_link,omitempty"`
	Status         string    `json:"status"`
	CreatedAt      time.Time `json:"created_at"`
	UpdatedAt      time.Time `json:"updated_at"`
}

type CreateApplicationRequest struct {
	ResumeID       string `json:"resume_id"`
	CompanyName    string `json:"company_name"`
	Role           string `json:"role"`
	RecruiterEmail string `json:"recruiter_email"`
	JobDescription string `json:"job_description"`
	JobLink        string `json:"job_link"`
}

type UpdateApplicationRequest struct {
	CompanyName    *string `json:"company_name,omitempty"`
	Role           *string `json:"role,omitempty"`
	RecruiterEmail *string `json:"recruiter_email,omitempty"`
	JobDescription *string `json:"job_description,omitempty"`
	JobLink        *string `json:"job_link,omitempty"`
}

// UpdateApplicationStatusRequest is the body for PATCH /api/applications/:id/status.
type UpdateApplicationStatusRequest struct {
	Status string `json:"status"`
}
